bgpd/bgp_bfd.c

```diff
@@ -283,6 +283,8 @@ static void bgp_bfd_peer_status_update(struct peer *peer, int status,
 
 	old_status = bfd_info->status;
 	bfd_info->status = status;
+	if (status == BFD_STATUS_ADMIN_DOWN)
+		bfd_info->status = BFD_STATUS_DOWN;
```

Comment:
this conversion is repeated in several places, but I don't quite get why it's always necessary? maybe it would be better to have a conversion macro or inline so that the pattern would be more like `xxx_info->status = BFD_CLIENT_STATUS(status)` or something like that?

Reply:
@mjstapp 
This is to ensure protocol like BGP, ISIS, OSPF always store BFD state as UP/DOWN
Do not want protocols to store more BFD states to keep it simple.

The idea of sending BFD_STATUS_ADMIN_DOWN is just for protocols to understand that protocol state need not go DOWN when BFD is ADMIN_DOWN. Protocol just stores the BFD state as DOWN in this case.

Reply:
so ... I think I understand what is being _done_, my comment is about whether there might be an alternative to the cut-and-paste way of doing it?

Reply:
will add as below, new_status value should not change since it is referenced in the code later.

#define BFD_SET_CLIENT_STATUS(current_status, new_status) 
        current_status = ((new_status == BFD_STATUS_ADMIN_DOWN)? BFD_STATUS_DOWN : new_status)